test/OpenTelemetry.Instrumentation.AspNetCore.Tests/BasicTests.cs

```diff
@@ -865,6 +865,61 @@ static void ThrowException(IApplicationBuilder app)
             await app.DisposeAsync();
         }
 
+        [Theory]
+        [InlineData(SamplingDecision.Drop)]
+        [InlineData(SamplingDecision.RecordOnly)]
+        [InlineData(SamplingDecision.RecordAndSample)]
+        public async Task ValidateContextExtraction(SamplingDecision samplingDecision)
+        {
+            try
+            {
+                var expectedTraceId = ActivityTraceId.CreateRandom();
+                var expectedParentSpanId = ActivitySpanId.CreateRandom();
+                var expectedTraceState = "rojo=1,congo=2";
+
+                DistributedContextPropagator.Current = new CustomDistributedContextPropagator(true, expectedTraceId, expectedParentSpanId, expectedTraceState);
```

Comment:
@vishweshbankwar Just FYI...

https://github.com/dotnet/aspnetcore/blob/442a380854c25b85252a743c3ba2e21f08fa6879/src/Hosting/Hosting/src/WebHostBuilder.cs#L292

https://github.com/dotnet/aspnetcore/blob/3ea008c80d5cc63de7f90ddfd6823b7b006251ff/src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs#L88

...it looks like AspNetCore registers `DistributedContextPropagator.Current` into the `IServiceCollection` and then uses DI to inject the propagator into its diagnostic code.

This test works because we're setting `DistributedContextPropagator.Current` before building the host. If you moved it below, it would probably fail.

Only bringing this up because we'll need to make sure we properly document this temporal coupling. It would be much simpler if AspNetCore just used the static :smile:

Reply:
Yes :) - I did notice that earlier that is why added the config at the top before build host. This is one of the things I wanted to discuss.  May be worth getting this updated in .NET8.0?